python/lsst/ctrl/bps/parsl/sites/ccin2p3.py

```diff
@@ -0,0 +1,248 @@
+                    # '#SBATCH' directives to prepend to the Slurm submission
+                    # script.
+                    scheduler_options=f"#SBATCH --qos={qos} --licenses=sps",
+                    # Set the number of file descriptors and processes to
+                    # the maximum allowed.
+                    worker_init="ulimit -n hard && ulimit -u hard",
+                    # Requests nodes which are not shared with other running
+                    # jobs.
+                    exclusive=False,
+                    # Should files be moved by Parsl?
+                    move_files=False,
+                ),
+                # Address to connect to the main Parsl process.
+                address=self.get_address(),
+                # GB of memory required per worker. If specified the node
+                # manager will check the available memory at startup and limit
+                # the number of workers such that the there’s sufficient memory
+                # for each worker.
+                mem_per_worker=None,
+                # Caps the number of workers launched per node.
+                max_workers=1,
+                # Timeout period (in milliseconds) to be used by the
+                # executor components.
+                poll_period=1_000,
+                # Retry submitting to Slurm in case of submission error.
+                block_error_handler=False,
+            )
+            executors.append(executor)
+
+        return executors
+
+    def select_executor(self, job: "ParslJob") -> str:
+        """Get the ``label`` of the executor to use to execute ``job``.
+
+        Parameters
+        ----------
+        job : `ParslJob`
+            Job to be executed.
+
+        Returns
+        -------
+        label : `str`
+            Label of executor to use to execute ``job``.
+        """
+        # We choose the executor to use based only on the memory required
+        # by the job.
+        memory = job.generic.request_memory / 1024  # Convert to GB
+        for label in ("small", "medium", "large"):
+            if memory <= self._slot_size[label]["memory"]:
+                return label
+
+        return "xlarge"
+
+    def get_parsl_config(self) -> parsl.config.Config:
+        """Get Parsl configuration for using CC-IN2P3 Slurm farm as a
+        Parsl execution site.
+
+        Returns
+        -------
+        config : `parsl.config.Config`
+            The configuration to be used to initialize Parsl for this site.
+        """
+        executors = self.get_executors()
+        monitor = self.get_monitor()
+        retries = get_bps_config_value(self.site, "retries", int, 1)
+        # Path to Parsl run directory. The default set by Parsl is
+        # 'runinfo' which is not explicit enough for end users given that
+        # we are using BPS + Parsl + Slurm to execute a workflow.
+        run_dir = get_bps_config_value(self.site, "run_dir", str, "parsl_runinfo")
```

Comment:
Oh, this is a nice feature! It might be good to hoist this into the parent class.